torch/functional.py

```diff
@@ -621,10 +625,39 @@ def stft(input: Tensor, n_fft: int, hop_length: Optional[int] = None,
                Note that calling :func:`torch.view_as_real` on the output will
                recover the deprecated output format.
 
+            .. deprecated:: 1.15.0
```

Comment:
Let's move the deprecation and warning to a separate PR and have this PR just add the new functionality